File: examples/db/create/node/createARecord.js
const { DbService } = require("m3o/db");

// Create a record in the database. Optionally include an "id" field otherwise it's set automatically.
async function createArecord() {
  let dbService = new DbService(process.env.MICRO_API_TOKEN);
  let rsp = await dbService.create({
    record: {
      age: 42,
      id: "1",
      isActive: true,
      name: "Jane",
    },
    table: "users",
  });
  console.log(rsp);
}

createArecord();

File: examples/db/create/go/createARecord.go
package example

import (
	"fmt"
	"os"

	"github.com/micro/services/clients/go/db"
)

// Create a record in the database. Optionally include an "id" field otherwise it's set automatically.
func CreateArecord() {
	dbService := db.NewDbService(os.Getenv("MICRO_API_TOKEN"))
	rsp, err := dbService.Create(&db.CreateRequest{
		Record: map[string]interface{}{
			"isActive": true,
			"id":       "1",
			"name":     "Jane",
			"age":      42,
		},
		Table: "users",
	})
	fmt.Println(rsp, err)
}
